tests/std/tests/P2445R1_forward_like/test.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <type_traits>
+#include <utility>
+
+using namespace std;
+
+struct U {}; // class type so const-qualification is not stripped from a prvalue
+using CU = const U;
+using T  = int;
+using CT = const T;
+
+U u{};
+const U& cu = u;
+
+static_assert(is_same_v<decltype(forward_like<T>(U{})), U&&>);
+static_assert(is_same_v<decltype(forward_like<T>(CU{})), CU&&>);
+static_assert(is_same_v<decltype(forward_like<T>(u)), U&&>);
+static_assert(is_same_v<decltype(forward_like<T>(cu)), CU&&>);
+static_assert(is_same_v<decltype(forward_like<T>(std::move(u))), U&&>);

```

Comment:
Note to other reviewers: I'm not requesting a comment clarifying why we qualify `std::move` here despite `using namespace std`. Yes, it's unconventional, but I assume this usage will become widespread when we adopt Clang 15 with the new warning about using `move` unqualified. (No change requested.)

Reply:
This warning might be obnoxious/widespread enough that we'll need to silence it in the matrices. (I'm not too worried about omitting `_STD` in product code, we're pretty good about that.) Still, no objections to doing this for now.